qiskit/optimization/algorithms/cobyla_optimizer.py

```diff
@@ -0,0 +1,148 @@
+
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The COBYLA optimizer wrapped to be used within Qiskit Optimization."""
+
+from typing import Optional
+
+import numpy as np
+from scipy.optimize import fmin_cobyla
+
+from qiskit.optimization.algorithms import OptimizationAlgorithm, OptimizationResult
+from qiskit.optimization.problems import QuadraticProgram, Constraint
+from qiskit.optimization import QiskitOptimizationError, infinity
+
+
+class CobylaOptimizer(OptimizationAlgorithm):
+    """The COBYLA optimizer wrapped to be used within Qiskit Optimization.
```

Comment:
How about the  'The SciPy COBYLA optimizer wrapped....' otherwise with this itself being called a COBYLA optimizer it seems a bit confusing perhaps